### DISCUSSION TITLE
MathML <mathml> alignment

### DISCUSSION THREAD

**Discussion by desaisam** [dita-ot/dita-ot#3929] [Q&A]
Team,

I am trying to use `<mathml>` (Inside `<equation-block>`)  in my DITA files and OT renders the the PDF accurately. However, I have one problem  By default its always 
"left" aligned. Can you suggest me a way to align the equations to left, right and center ?

Thanks in advance

**Comment by raducoravu** [comment]
Do you mean that the entire equation is placed on the left part of the page and you want for example to have it in the center? There is no out of the box solution for this.
You would probably first need to place it in a block element like a paragraph or a `<fig>` element with a certain @outputclass attribute value:
```xml
      <fig outputclass="right-align">
        <mathml>
          <m:math xmlns:m="http://www.w3.org/1998/Math/MathML">
            <m:mi>C</m:mi>
            <m:msub>
              <m:mi>O</m:mi>
              <m:mn>2</m:mn>
            </m:msub>
          </m:math>
        </mathml>
      </p>
```
If you produce the PDF using the default PDF processing bundled in the DITA OT then you probably need to create a PDF customization which using XSLT matches on such paragraphs/figures and creates `fo:block` elements with `text-align="right"`:

https://www.dita-ot.org/dev/topics/pdf-customization.html

If you produce the PDF using a commercial CSS based processor like the the Oxygen Chemistry engine, you need to create a small custom CSS to specify the alignment:

https://www.oxygenxml.com/doc/versions/24.1/ug-editor/topics/dcpp_images.html#ariaid-title8

something like:
```css
*[class ~= "right-align"] {
  text-align:right;
}
```
